ssz_serialization/types.nim

```diff
@@ -623,8 +623,8 @@ template writeValue*(writer: var JsonWriter, value: List) =
   else:
     writeValue(writer, asSeq value)
 
-proc writeValue*(writer: var JsonWriter, value: HashList)
-                {.raises: [IOError, SerializationError].} =
+proc writeValue*(
+    writer: var JsonWriter, value: HashList) {.raises: [IOError].} =
```

Comment:
nim-ser generally only raises `SerializationError` on `readValue`, `writeValue` never raises `SerializationError`, if a `writeValue` implementation is given it is expected that it can serialize the entire range of input values.

Reply:
This asymetry makes sense, actually. When reading an unsanitized value, it can be invalid. While, on writing, it can't be, as the type safety guarantees all values to be valid. IOErrors can happen both ways though.

Reply:
>  While, on writing, it can't be, as the type safety guarantees all values to be valid

unless the value cannot be serialized - ie due to range issues or whatever other constraints the output format places on serialized data that may not necessarily be in tune with the Nim type, if using arbitrary nim types. I dunno, it's a future option that might come in handy but I don't feel too strongly about it either.

Reply:
- `nim-serialization` does not synthesize `SerializationError` when it creates implicit implementations
- `nim-chronicles` does not try and deal with `SerializationError` when logging
- The concept is weird to have objects that can't be serialized. Just convert to a representation that always succeeds, or ensure in the caller that no illegal values are serialized.

Reply:
etan has a point - ie there are alternative solutions to this and the cost of having to deal with yet another exception should perhaps not be underestimated

Reply:
Well, it's also simply not how `nim-serialization` currently works: It only has `IOError` on `writeValue`. For only a handful of handlers we accidentally copy pasted the `SerializationError` from the `readValue` when implementing `writeValue`. None of them actually raises it, it's a simple oversight.

So, if we _want_ `SerializationError` to be raisable from `writeValue`, that would be a new feature. We shouldn't use the copy paste accidents as precedence to implement it, though.

Reply:
Another instance of accidental `SerializationError`: 
- https://github.com/status-im/nim-eth/pull/631

Here, `{.raises: [SerializationError].}` gets unintentionally `{.push.}`-ed to `writeValue` implementations, instead of just `readValue` implementations.